plugins/org.jboss.tools.openshift.reddeer/src/org/jboss/tools/openshift/reddeer/requirement/OpenShiftProjectRequirement.java

```diff
@@ -0,0 +1,144 @@
+import org.jboss.tools.openshift.core.connection.Connection;
+import org.jboss.tools.openshift.reddeer.condition.OpenShiftProjectExists;
+import org.jboss.tools.openshift.reddeer.requirement.OpenShiftProjectRequirement.RequiredProject;
+import org.jboss.tools.openshift.reddeer.utils.DatastoreOS3;
+
+import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.model.IProject;
+import com.openshift.restclient.model.project.IProjectRequest;
+
+/**
+ * A requirement that makes sure a given project exists. If it doesnt it will get created
+ * 
+ * @author adietish@redhat.com
+ */
+public class OpenShiftProjectRequirement implements Requirement<RequiredProject> {
+
+	private final static Logger LOGGER = new Logger(OpenShiftProjectRequirement.class);
+
+	private RequiredProject projectSpec;
+
+	private IProject project;
+
+	@Retention(RetentionPolicy.RUNTIME)
+    @Target(ElementType.TYPE)
+    public @interface RequiredProject {
+		/**
+		 * the connection to use when connecting. If nothing is provided
+		 * {@link DatastoreOS3#SERVER} and {@link DatastoreOS33#USERNAME} are
+		 * used.
+		 */
+		String connectionURL() default StringUtils.EMPTY;
+
+		/**
+		 * the project name to use. If nothing is provided
+		 * {@link OpenShiftResources#TEST_PROJECT} is used
+		 */
+		String name() default StringUtils.EMPTY;
+		/**
+		 * the project display name to use. If nothing is provided
+		 * {@link #name()} is used
+		 */
+		String displayName() default StringUtils.EMPTY;
+		/**
+		 * the project description to use. If nothing is provided
+		 * {@link #name()} is used
+		 */
+		String description() default StringUtils.EMPTY;
+	}
+	
+	@Override
+	public boolean canFulfill() {
+		return true;
+	}
+
+	@Override
+	public void fulfill() {
+		String projectName = DatastoreOS3.getValueOrDefault(projectSpec.name(), DatastoreOS3.TEST_PROJECT);
+		Connection connection = ConnectionUtils.getConnectionOrDefault(projectSpec.connectionURL());
+		assertNotNull(NLS.bind("No connection for {0} exists", projectSpec.connectionURL()), connection);
+
+		this.project = getOrCreateProject(projectName, connection);
+	}
+
+
+	private IProject getOrCreateProject(String name, Connection connection) {
+		IProject project = OpenShiftResourceUtils.getProject(name, connection);
+		if (project == null) {
+			LOGGER.debug(NLS.bind("Project {0} doesnt exist yet in {1}, creating it.", name, connection.getHost()));
+			project = createProject(name, projectSpec, connection);
+//			new WaitUntil(
+//					new ProjectExists(name, connection),
+//					TimePeriod.LONG);
+			/**
+			 * WORKAROUND: explorer wont get notified of the the new project and
+			 * therefore wont display it unless a manual refresh is done on the connection.
+			 * https://issues.jboss.org/browse/JBIDE-23184 remove this wait once
+			 * WatchManager is watching projects and notifies the ui.
+			 * 
+			 * @see WatchManager#KINDS
+			 */
+			new WaitUntil(new OpenShiftProjectExists(name, connection));
```

Comment:
this wait (that refreshes the explorer/connection waits for the project to show up) fixes the issue. It should be remove once https://issues.jboss.org/browse/JBIDE-23184 is resolved